.github/workflows/tests.yml

```diff
@@ -2,48 +2,50 @@ name: run tests
 
 on:
   push:
+  pull_request:
   schedule:
-    - cron: '* * * * 0'  # run once a week (sunday)
+    - cron: '0 12 * * 0'  # run once a week on Sunday
 
 jobs:
   build:
     strategy:
       matrix:
-        python-version:
-        - "2.7"
-        - "3.5"
-        - "3.6"
-        - "3.7"
-        - "3.8"
-        - "3.9"
-        - "pypy2"
-        - "pypy3"
-        include:
-          - python-version: "3.8"
-            env:
-              - lint
-              - coverage
+        config:
+        # [name, Python version, tox env]
+        - ["lint",     "3.8",   "lint"]
+        - ["2.7",      "2.7",   "py27"]
+        - ["3.5",      "3.5",   "py35"]
+        - ["3.6",      "3.6",   "py36"]
+        - ["3.7",      "3.7",   "py37"]
+        - ["3.8",      "3.8",   "py38"]
+        - ["3.9",      "3.9",   "py39"]
+        - ["pypy2",    "pypy2", "pypy"]
+        - ["pypy3",    "pypy3", "pypy3"]
+        - ["coverage", "3.8",   "coverage"]
 
     runs-on: ubuntu-latest
+    name: ${{ matrix.config[0] }}
     steps:
     - uses: actions/checkout@v2
     - name: Set up Python
       uses: actions/setup-python@v2
       with:
-        python-version: ${{ matrix.python-version }}
+        python-version: ${{ matrix.config[1] }}
     - name: Pip cache
       uses: actions/cache@v2
       with:
         path: ~/.cache/pip
-        key: ${{ runner.os }}-pip-${{ hashFiles(setup.*', 'tox.ini') }}
+        key: ${{ runner.os }}-pip-${{ hashFiles('setup.*', 'tox.ini') }}
         restore-keys: |
           ${{ runner.os }}-pip-
     - name: Install dependencies
       run: |
         python -m pip install --upgrade pip
-        pip install coveralls
+        pip install tox coveralls
     - name: Test
-      run: tox -e {{ matrix.env || 'py'}}
+      run: tox -e ${{ matrix.config[2] }}
     - name: Coverage
+      if: matrix.config[1] == 'coverage'
```

Comment:
Off-by-one error.  The matrix config is:

    ["coverage", "3.8",   "coverage"]

So it should be either `matrix.config[0]` or `matrix.config[2]`, otherwise it never matches.

Reply:
I was not sure whether it is zero or one based. And I forgot to check. I am going to fix it in #9.

Reply:
Thank you for bringing this up.